message_definitions/v1.0/common.xml

```diff
@@ -400,43 +400,49 @@
       </entry>
       <entry value="4" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_ROLL_AXIS">
         <description>Gimbal device supports rotating around roll axis.</description>
       </entry>
       <entry value="8" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_ROLL_FOLLOW">
-        <description>Gimbal device supports to follow a roll angle relative to the vehicle</description>
+        <description>Gimbal device supports to follow a roll angle relative to the vehicle.</description>
       </entry>
       <entry value="16" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_ROLL_LOCK">
-        <description>Gimbal device supports locking to an roll angle (generally that's the default with roll stabilized)</description>
+        <description>Gimbal device supports locking to a roll angle (generally that's the default with roll stabilized).</description>
       </entry>
       <entry value="32" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_PITCH_AXIS">
         <description>Gimbal device supports rotating around pitch axis.</description>
       </entry>
       <entry value="64" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_PITCH_FOLLOW">
-        <description>Gimbal device supports to follow a pitch angle relative to the vehicle</description>
+        <description>Gimbal device supports to follow a pitch angle relative to the vehicle.</description>
       </entry>
       <entry value="128" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_PITCH_LOCK">
-        <description>Gimbal device supports locking to an pitch angle (generally that's the default with pitch stabilized)</description>
+        <description>Gimbal device supports locking to a pitch angle (generally that's the default with pitch stabilized).</description>
       </entry>
       <entry value="256" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_YAW_AXIS">
         <description>Gimbal device supports rotating around yaw axis.</description>
       </entry>
       <entry value="512" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_YAW_FOLLOW">
-        <description>Gimbal device supports to follow a yaw angle relative to the vehicle (generally that's the default)</description>
+        <description>Gimbal device supports to follow a yaw angle relative to the vehicle (generally that's the default).</description>
       </entry>
       <entry value="1024" name="GIMBAL_DEVICE_CAP_FLAGS_HAS_YAW_LOCK">
-        <description>Gimbal device supports locking to an absolute heading (often this is an option available)</description>
+        <description>Gimbal device supports locking to an absolute heading, i.e., yaw angle relative to North (earth frame, often this is an option available).</description>
       </entry>
       <entry value="2048" name="GIMBAL_DEVICE_CAP_FLAGS_SUPPORTS_INFINITE_YAW">
         <description>Gimbal device supports yawing/panning infinetely (e.g. using slip disk).</description>
       </entry>
+      <entry value="4096" name="GIMBAL_DEVICE_CAP_FLAGS_SUPPORTS_YAW_IN_EARTH_FRAME">
```

Comment:
Hi, thanks for all this.

What's the difference between SUPPORT_YAW_IN_EARTH_FRAME and HAS_YAW_LOCK?

Reply:
I think SUPPORT_YAW_IN_EARTH_FRAME refers to if it accepts AUTOPILOT_STATE_FOR_GIMBAL_DEVICE from the autopilot, while HAS_YAW_LOCK refers to gimbal device capability of doing yaw lock or yaw follow. Not 100% sure though, I just brought it back as it is from mavlink/mavlink.

Reply:
OK, that's a very odd name for the capability if that's its real purpose.  In any case, as you say this is already upstream so I don't mind it coming into AP.